components/sup/src/main.rs

```diff
@@ -173,7 +176,15 @@ fn sub_run_rsr_imlw_mlw_gsw_smw_rhw_msw(m: &ArgMatches,
     set_supervisor_logging_options(m);
 
     let cfg = mgrcfg_from_sup_run_matches(m, feature_flags)?;
-    let manager = Manager::load_imlw(cfg, launcher)?;
+
+    let maybe_sys_ip = m.value_of("SYS_IP_ADDRESS")
+                        .and_then(|s| IpAddr::from_str(s).ok());
+    let sys_ip = maybe_sys_ip.unwrap_or_else(|| {
+                     habitat_core::util::sys::ip().unwrap_or(IpAddr::V4(Ipv4Addr::LOCALHOST))
+                 });
```

Comment:
OPTIONAL: This can be simplified and made easier to read and understand:
```rust
let sys_ip = m.value_of("SYS_IP_ADDRESS")
     .and_then(|s| IpAddr::from_str(s).ok())
     .or_else(|| habitat_core::util::sys::ip().ok())
     .unwrap_or_else(|| IpAddr::V4(Ipv4Addr::LOCALHOST));
```
and if we're willing to change `habitat_core::util::sys::ip` to return an `Option` (this is its only caller, so I see no reason not to), we can simplify even further:
```rust
let sys_ip = m.value_of("SYS_IP_ADDRESS")
              .and_then(|s| IpAddr::from_str(s).ok())
              .or_else(habitat_core::util::sys::ip)
              .unwrap_or_else(|| IpAddr::V4(Ipv4Addr::LOCALHOST));
```